src/components/map/MapLibre.tsx

```diff
@@ -0,0 +1,315 @@
+  const { alternativePosition, defaultAlternativePosition } = locationSettings;
+  const [followsUserLocation, setFollowsUserLocation] = useState(false);
+  const [selectedFeature, setSelectedFeature] = useState(null);
+  const mapRef = useRef(null);
+  const cameraRef = useRef(null);
+  const shapeSourceRef = useRef<ShapeSourceRef>(null);
+
+  let initialRegion: LocationObjectCoords = {
+    latitude: 51.1657,
+    longitude: 10.4515
+  };
+
+  if (locations?.[0]?.position?.latitude && locations[0]?.position?.longitude) {
+    initialRegion = {
+      ...initialRegion,
+      latitude: locations[0].position.latitude,
+      longitude: locations[0].position.longitude
+    };
+  }
+
+  let mapCenterPosition = {} as LocationObjectCoords;
+
+  if (alternativePosition) {
+    mapCenterPosition = getLocationMarker(alternativePosition).position;
+  } else if (defaultAlternativePosition) {
+    mapCenterPosition = getLocationMarker(defaultAlternativePosition).position;
+  }
+
+  if (mapCenterPosition) {
+    initialRegion = {
+      ...initialRegion,
+      ...mapCenterPosition
+    };
+  }
+
+  const handleClusterPress = async (event: any) => {
+    const cluster = event.features[0];
+
+    if (cluster?.type === 'Feature') {
+      const zoomForCluster = (await shapeSourceRef.current?.getClusterExpansionZoom(cluster)) ?? 0;
+      const newZoomLevel = Math.min(zoomForCluster, 20);
+
+      cameraRef.current?.setCamera({
+        animationDuration: 1500,
+        animationMode: 'flyTo',
+        centerCoordinate: cluster.geometry.coordinates,
+        zoomLevel: newZoomLevel
+      });
+    }
+  };
+
+  return (
+    <View style={[styles.container, style]}>
+      <MapView
+        attributionEnabled={false}
+        compassEnabled={false}
+        mapStyle="https://tileserver-gl.smart-village.app/styles/osm-liberty/style.json"
+        ref={mapRef}
+        rotateEnabled={false}
+        style={[styles.map, mapStyle]}
+        zoomEnabled
+      >
+        <Camera
+          centerCoordinate={[initialRegion.longitude, initialRegion.latitude]}
+          followUserLocation={followsUserLocation}
+          followUserMode={UserTrackingMode.FollowWithHeading}
+          minZoomLevel={minZoom}
+          ref={cameraRef}
+          zoomLevel={initialZoomLevel}
+        />
+
+        <UserLocation
+          androidRenderMode="compass"
+          renderMode={UserLocationRenderMode.Native}
+          showsUserHeadingIndicator
+          visible={showsUserLocation}
+        />
+
+        <Images images={markerImages} />
+
+        <ShapeSource
+          id="benches"
+          ref={shapeSourceRef}
+          shape={featureCollection(
+            locations?.map((location) =>
+              point([location.position.longitude, location.position.latitude], { ...location })
+            )
+          )}
+          onPress={handleClusterPress}
+          cluster
+          clusterRadius={50}
```

Comment:
since we have a config file. Then I think we should also add the `clusterRadius` value to the config.

Reply:
of course. there will be even more, the map style url for example